### PR TITLE
Added tests for Pipe and Process

### LINKED ISSUE #1220: Write tests for Pipe and Process
These classes are never tested.

- [ ] `Pipe` (including of course `PipeUnix`)
- [ ] `Process` (including of course `ProcessUnix`)

### PR DESCRIPTION
**Pull request by marcusnaslund**
Fixes #1220

### CODE REVIEW

**Comment by marcusnaslund** [comment]
Peer review @sebastianbaginski ?

File: test/io/ProcessTest.ooc [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+			scriptArgs free()
+
+			reader := FileReader new(t"test/io/output/sum.txt")
+			expect(reader hasNext())
+			data := CString new(16)
+			reader read(data, 0, 15)
+			string := data toString()
+			expect(string, is equal to("50005000 abcABC"))
+			string free()
+			memfree(data)
```
Comment (sebastianbaginski) [inline test/io/ProcessTest.ooc]: `CString` does not have a `free` function ?

File: test/io/input/pipeprocesstester.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+sum_to () (
+    set -- $(seq $1)
+    IFS=+
```
Comment (sebastianbaginski) [inline test/io/input/pipeprocesstester.sh]: When a bash function is changing the internal field separator then I think it does something more than simply `sum_to` :)

File: test/io/ProcessTest.ooc (hunk 1 above)
Comment (marcusnaslund) [inline test/io/ProcessTest.ooc]: No, but it's only a `Char*` cover.

**Comment by sebastianbaginski** [comment]
OK, nice test, looks good.

File: test/io/input/pipeprocesstester.sh (hunk 2 above)
Comment (marcusnaslund) [inline test/io/input/pipeprocesstester.sh]: I actually have no idea what this is doing, I got it online. Bash is just awful.

Comment (sebastianbaginski) [inline test/io/input/pipeprocesstester.sh]: What do you want it to do ? :)

Comment (sebastianbaginski) [inline test/io/input/pipeprocesstester.sh]: I think it sums the numbers by redirecting the string created from numeric values separated by `+` to `bc`. So it calculates `1+2+3+4+5+..` by concatenating and sending this string to separate process to do the calculations :D

Comment (marcusnaslund) [inline test/io/input/pipeprocesstester.sh]: I just wanted some kind of function that takes some time to run. My only critera was that it does not contain `rm -rf /`. This did the trick. :)

Comment (sebastianbaginski) [inline test/io/input/pipeprocesstester.sh]: You can simply use sleep - `sleep 0.5` will delay for half second.

Comment (marcusnaslund) [inline test/io/input/pipeprocesstester.sh]: Oh. Ah well. If this doesn't do anything bad I'm happy this way.

**Pull request merged**